.github/workflows/code-quality.yml

```
@@ -0,0 +1,196 @@
+          DB_CONNECTION: sqlite
+          DB_DATABASE: ':memory:'
+      
+      - name: Generate test coverage report
+        if: success()
+        run: |
+          php artisan test --coverage --coverage-html=coverage
+          php artisan test --coverage-text
+      
+      - name: Upload coverage to Codecov
+        if: success()
+        uses: codecov/codecov-action@v3
```

actionlint (1.7.7)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)